loopy/kernel/tools.py

```diff
@@ -1415,6 +1415,14 @@ def conform_to_uniform_length(s):
 
 # {{{ stringify_instruction_list
 
+def stringify_instruction_tag(tag):
+    from loopy.kernel.instruction import LegacyStringInstructionTag
+    if isinstance(tag, LegacyStringInstructionTag):
```

Comment:
Why not move this to `LegacyStringInstructionTag.__str__`?

Reply:
I actually like the default dataclass `__str__` just fine. It's just a tad wordy, especially in this setting, where fitting a lot of information on the screen is important.